packages/components/pie-toast-provider/src/index.ts

```diff
@@ -1,20 +1,143 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-import { RtlMixin, defineCustomElement } from '@justeattakeaway/pie-webc-core';
-
+import {
+    LitElement,
+    html,
+    nothing,
+    unsafeCSS,
+    type PropertyValues,
+} from 'lit';
+import { state, property } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import {
+    RtlMixin,
+    defineCustomElement,
+    dispatchCustomEvent,
+} from '@justeattakeaway/pie-webc-core';
+import '@justeattakeaway/pie-toast';
 import styles from './toast-provider.scss?inline';
-import { type ToastProviderProps } from './defs';
+import {
+    defaultProps,
+    PRIORITY_ORDER,
+    type ToastProviderProps,
+    type ExtendedToastProps,
+    ON_TOAST_PROVIDER_QUEUE_UPDATE_EVENT,
+} from './defs';
 
 // Valid values available to consumers
 export * from './defs';
+export { toaster } from './toaster';
 
 const componentSelector = 'pie-toast-provider';
 
 /**
  * @tagname pie-toast-provider
+ * @event {CustomEvent} pie-toast-provider-queue-update - when a toast is added or removed from the queue.
  */
 export class PieToastProvider extends RtlMixin(LitElement) implements ToastProviderProps {
+    @state()
+    private _toasts: ExtendedToastProps[] = [];
+
+    @state()
+    private _currentToast: ExtendedToastProps | null = null;
```

Comment:
If we still need these `ExtendedToastProps`, should they live inside the toast component instead of the provider?

Reply:
It feels like this line should say
```typescript
private _currentToast: PieToast | null = null;
```
and same for `_toasts`; it feels like that should probably be of type `PieToast[]`.

Reply:
we need to extend the `PieToast` interface to include the events so they can be passed to the `createToast` method, otherwise the events won't be recognised by typescript and we don't have a way to trigger/pass the events to the toast component. 

https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-toast-provider/src/index.ts#L134-L136

```
createToast({
    onPieToastClose: () => log(),
    onPieToastOpen: () => log(),
    onPieToastLeadingActionClick: () => log()
})

```